src/main/java/roomescape/repository/TimeDAO.java

```diff
@@ -0,0 +1,61 @@
+package roomescape.repository;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+import roomescape.domain.Time;
+
+@Repository
+public class TimeDAO {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    public TimeDAO(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    public Time insert(Time time) {
+        String sql = "INSERT INTO time (time) VALUES ?";
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+
+        jdbcTemplate.update(connect -> {
+            PreparedStatement ps = connect.prepareStatement(
+                    sql,
+                    new String[]{"id"});
+            ps.setString(1, time.getTime());
+            return ps;
+        }, keyHolder);
+
+        return findById(keyHolder.getKey().longValue()).get();
+    }
```

Comment:
```suggestion
    public Time insert(Time time) {
        String sql = "INSERT INTO time (time) VALUES ?";
        KeyHolder keyHolder = new GeneratedKeyHolder();

        jdbcTemplate.update(connect -> {
            PreparedStatement ps = connect.prepareStatement(
                    sql,
                    new String[]{"id"});
            ps.setString(1, time.getTime());
            return ps;
        }, keyHolder);

        Long generatedId = keyHolder.getKey().longValue();
        return new Time(generatedId, time.getTime());
    }
```

- `generatedId`가 정상적으로 반환되었다는 것은 잘 삽입되었다는 것이니 id만 활용해서 바로 객체를 만들고 반환하면 `findById()` 메서드를 사용하지 않아도 될 것 같아요! 소정의 생각은 어떤가요?

Reply:
@sojeong0202 다시 질문하신거 답변했어요!

Reply:
그러네요, 이 메소드 주요 목적인 DB 삽입은 잘 되었으니 
굳이 DB에서 찾아 가져올 필요 없이 keyHolder와 삽입에 사용한 `Time` 객체의 필드 값으로 새로운 객체를 만들어 반환하면 되겠군요!

좋은 방법 알려주셔서 감사합니다!! 수정했습니다